lib/bin/gpa-games-cli

```diff
@@ -67,8 +67,7 @@ class HexCliGameState(UiGameState):
             raise("Unrecognized player, \"{}\"".format(ui_player))
 
     def reset_state(self):
-        self.state.reset(*self.state.board.size())
-        return self
+        self.state.reset()
```

Comment:
I guess my concern here is that a function with side-effects which returns the argument it mutates doesn't seem very idiomatic. I can't seem to find anything that explicitly points this out, but I believe it's usually the case in Python that either a function mutates its arguments and returns None, or else it creates and returns a mutated copy based on the immutable arguments.

For example, consider the difference between List.sort() and the built-in sorted(). List.sort() mutates the list in place and returns None, while sorted() returns a sorted copy of the list. Another example would be methods of str, which are often chained, but explicitly provide a copy of the string instead of mutating the argument string when called.

Having a function which both mutates its argument and returns it could be confusing if  you're not familiar with how the function breaks the regular contract. Someone might try to do something like:

```
state_copy = state.reset_state() # Take a copy of the state.
do_something(state)
do_something_else(state_copy)
```

Without realising that what was returned from state.reset_state() was the same state object, and not a copy as is usually the case for functions with side-effects that return something.

Is there a reason why method chaining would be preferable compared to calling each method by itself?

```
state.reset_state().mutate()
```

Seems less explicit than:

```
state.reset_state()
state.mutate()
```